htdocs/salaries/card.php

```diff
@@ -738,6 +738,21 @@ function( data ) {
 		$formquestion[] = array('type' => 'text', 'name' => 'amount', 'label' => $langs->trans("Amount"), 'value' => price($object->amount), 'morecss' => 'width100 right');
 
 		$formconfirm = $form->formconfirm($_SERVER["PHP_SELF"].'?id='.$object->id, $langs->trans('ToClone'), $langs->trans('ConfirmCloneSalary', $object->ref), 'confirm_clone', $formquestion, 'yes', 1, 280);
+
+		//Add fill with end of month button
+		$formconfirm .= "<script>
+			$('#clone_date_end').after($('<button id=\"fill_end_of_month\" class=\"dpInvisibleButtons\" style=\"color: var(--colortextlink);font-size: 0.8em;opacity: 0.7;margin-left:4px;\" type=\"button\">". $langs->trans('FillEndOfMonth')."</button>'));
+			$('#fill_end_of_month').click(function(){
+				var clone_date_startmonth = +$('#clone_date_startmonth').val();
+				var clone_date_startyear = +$('#clone_date_startyear').val();
+				var end_date = new Date(clone_date_startyear, clone_date_startmonth, 0);
+				end_date.setMonth(clone_date_startmonth - 1);
+				$('#clone_date_end').val( (new Intl.DateTimeFormat('".str_replace("_", "-", $langs->shortlang)."',{year: 'numeric',month: '2-digit',day: '2-digit'})).format(end_date));
```

Comment:
We are on the good way. Still one thing to change. Here you use the country of user to format the visible string of date.
A better solution is to use the string format that is related to the active user language and found with
$langs->trans("FormatDateShortJavaInput")
Just replace the string %Y with year on 4 char, %m with month on 2 char, %d with day on 2 char in this string and you will get the string to inject into $("#clone_date_end)
Si we are using the same method than into the lethod selectDate() used to output the date selector.

Reply:
Good, I found a function formatDate(date,format) into core\js\lib_head.js.php it return the needed output format.